api/hasura/actions/upload_profile_avatar.ts

```diff
@@ -0,0 +1,96 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { v4 as uuidv4 } from 'uuid';
+
+import { gql } from '../../../api-lib/Gql';
+import { deleteImage, uploadImage } from '../../../api-lib/s3';
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  updateProfileAvatarInput,
+} from '../../../src/lib/zod';
+import { parseBase64Image, resizeAvatar } from '../images';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  // parse the input
+  const {
+    input: { object: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    updateProfileAvatarInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+
+  try {
+    const imageBytes = parseBase64Image(res, input.image_data_base64);
+    if (!imageBytes) {
+      return res;
```

Comment:
should this return some form of a 422 error?

Reply:
It will either return the imageBytes, or a 413, as far as I can see.